rcl/test/rcl/test_graph.cpp

```diff
@@ -1321,3 +1330,172 @@ TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION), test_rcl_service_server_
+  test_rcl_get_publishers_info_by_topic_null_allocator)
+{
+  auto ret = rcl_get_publishers_info_by_topic(this->node_ptr, nullptr, this->topic_name, false,
+      this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_allocator)
+{
+  auto ret = rcl_get_subscriptions_info_by_topic(this->node_ptr, nullptr, this->topic_name, false,
+      this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_null_topic)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  auto ret = rcl_get_publishers_info_by_topic(this->node_ptr,
+      &allocator, nullptr, false, this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_topic)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  auto ret = rcl_get_subscriptions_info_by_topic(this->node_ptr,
+      &allocator, nullptr, false, this->topic_info_array);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_null_participants)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  auto ret = rcl_get_publishers_info_by_topic(this->node_ptr,
+      &allocator, this->topic_name, false, nullptr);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_subscriptions_info_by_topic_null_participants)
+{
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  auto ret = rcl_get_subscriptions_info_by_topic(this->node_ptr,
+      &allocator, this->topic_name, false, nullptr);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+}
+
+/*
+ * This does not test content of the response.
+ * It only tests if the return code is the one expected.
+ */
+TEST_F(CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION),
+  test_rcl_get_publishers_info_by_topic_invalid_participants)
+{
+  // this participant is invalid as the pointer "participants" inside is expected to be null.
+  this->topic_info_array->info_array =
+    reinterpret_cast<rmw_topic_info_t *>(malloc(sizeof(rmw_topic_info_t *)));
```

Comment:
is this correct?
Its allocating a pointer of size (rmw_topic_info_t pointer size).
Did you mean:
```c++
reinterpret_cast<rmw_topic_info_t *>(malloc(sizeof(rmw_topic_info_t)));
```

This is a personal nit, but I'd use `calloc` for tests since it zeroes out the allocated memory where as `malloc` is allowed to allocate chunks of garbage. 

I'd say the performance cost is worth consistency, especially in tests.